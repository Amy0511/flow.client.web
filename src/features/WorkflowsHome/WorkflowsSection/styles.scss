@import "~@boomerang/boomerang-styles/lib/colors";

.c-workflow-section {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: auto;
  margin-top: 3rem;
  .bmrg--b-button {
    margin-right: 1rem;
  }

  .bmrg--b-tooltip.--bmrg-white {
    border: 0rem;
    margin-top: 0.625rem;
  }
}
.c-workflow-section__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 1rem;
}
.b-workflow-section__team {
  margin-left: 1rem;
  font-size: 1rem;
  font-weight: 600;
}
.b-workflow-section__button {
  margin-right: 1rem;
}
.c-workflow-section__workflows {
  display: flex;
  flex-wrap: wrap;
}

.b-workflow-placeholder {
  border: none;
  background: none;
  cursor: pointer;
  margin-left: 1rem;
}

.b-workflow-placeholder__box {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20rem;
  height: 9.325rem;
  border-radius: 0.25rem;
  border: 1px dashed $cool-gray-20;
}

.b-workflow-placeholder__text {
  font-size: 4rem;
  line-height: 4rem;
  font-weight: 200;
  color: $cool-gray-50;
}
