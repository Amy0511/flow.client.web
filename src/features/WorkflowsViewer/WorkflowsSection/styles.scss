.c-workflow-section {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: auto;
  margin-top: 3rem;
  .bmrg--b-button{
    margin-right: 1rem;
  }
}
.c-workflow-section__header{
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 1rem;
}
.b-workflow-section__team{
  margin-left: 1rem;
  font-size: 1rem;
  font-weight: 600;
}
.b-workflow-section__button{
  margin-right: 1rem;
}
.c-workflow-section__workflows{
  display: flex;
  flex-wrap: wrap;
}

