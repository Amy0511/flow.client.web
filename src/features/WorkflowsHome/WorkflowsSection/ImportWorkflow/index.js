import React, { Component } from "react";
import PropTypes from "prop-types";
import Modal from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import { Icon } from "carbon-components-react";
import ImportAttachment from "./ImportAttachment";
import ImportConfirm from "./ImportConfirm";
import ImportResult from "./ImportResult";
import ImportType from "./ImportType";
import "./styles.scss";

class ImportWorkflow extends Component {
  static propTypes = {
    importWorkflow: PropTypes.object.isRequired,
    fetchTeams: PropTypes.func.isRequired,
    handleImportWorkflow: PropTypes.func.isRequired
  };

  render() {
    const initialState = {
      step: 0,
      formData: {
        files: [],
        isUpdate: false
      }
    };

    return (
      <Modal
        modalProps={{ shouldCloseOnOverlayClick: false }}
        ModalTrigger={() => <Icon className="b-workflow-import__icon" name="icon--upload" alt="Import Workflow" />}
        initialState={initialState}
        theme="bmrg-white"
        modalContent={(closeModal, rest) => (
          <ModalFlow
            headerTitle="Workflow Import"
            headerSubtitle="Import your own workflow"
            closeModal={closeModal}
            confirmModalProps={{ affirmativeAction: closeModal, theme: "bmrg-white" }}
            theme="bmrg-white"
            {...rest}
          >
            <ImportType />
            <ImportAttachment />
            <ImportConfirm handleImportWorkflow={this.props.handleImportWorkflow} />
            <ImportResult importWorkflow={this.props.importWorkflow} fetchTeams={this.props.fetchTeams} />
          </ModalFlow>
        )}
      />
    );
  }
}

export default ImportWorkflow;
