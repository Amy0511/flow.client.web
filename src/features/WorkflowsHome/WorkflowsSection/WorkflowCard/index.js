import React, { Component } from "react";
import axios from "axios";
import { notify, Notification } from "@boomerang/boomerang-components/lib/Notifications";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { OverflowMenu, OverflowMenuItem } from "carbon-components-react";
import { BASE_SERVICE_URL } from "Config/servicesConfig";
import Tooltip from "@boomerang/boomerang-components/lib/Tooltip";
import AlertModal from "@boomerang/boomerang-components/lib/AlertModal";
import ConfirmModal from "@boomerang/boomerang-components/lib/ConfirmModal";
import imgs from "Assets/icons";
import "./styles.scss";
import playButton from "./img/playButton.svg";

class WorkflowCard extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired,
    workflow: PropTypes.object.isRequired,
    updateWorkflows: PropTypes.func.isRequired
  };

  handleExecute = () => {
    return axios
      .post(`${BASE_SERVICE_URL}/execute/${this.props.workflow.id}`)
      .then(response => {
        notify(<Notification type="success" title="Run Workflow" message="Succssfully ran workflow" />);
      })
      .catch(error => {
        notify(<Notification type="error" title="Something's wrong" message="Failed to run workflow" />);
      });
  };

  handleOnDelete = () => {
    // axios
    //   .delete(`${BASE_SERVICE_URL}/workflow/${this.props.workflow.id}`)
    //   .then(() => {
    //     notify(<Notification type="remove" title="SUCCESS" message="Workflow successfully deleted" />);
    //     this.props.updateWorkflows({workflowId:this.props.workflow.id, teamId:this.props.teamId});
    //     return;
    //   })
    //   .catch(() => {
    //     notify(<Notification type="error" title="SOMETHING'S WRONG" message="Your delete request has failed" />);
    //     return;
    //   });
    //this.props.updateWorkflows({ workflowId: this.props.workflow.id, teamId: this.props.teamId });
  };

  render() {
    const { workflow, history } = this.props;
    const menuOptions = [
      {
        itemText: "View Activity",
        onClick: () => history.push(`/activity/dfb6302b-62e0-4574-9062-727e4a37fc32`),
        primaryFocus: true
      },
      {
        itemText: "Edit Workflow",
        onClick: () => history.push(`/editor/${workflow.id}/designer`),
        primaryFocus: false
      },
      { itemText: "Delete", onClick: () => this.handleOnDelete(), primaryFocus: false }
    ];

    return (
      <div className="c-workflow-card">
        <div className="c-workflow-card__header">
          <div className="c-workflow-card__status">
            <div className={`b-workflow-card__circle --${workflow.status}`} />
            <label className="b-workflow-card__status">{workflow.status}</label>
          </div>
          <OverflowMenu className="b-workflow-card__menu" ariaLabel="" iconDescription="">
            {menuOptions.map(option => {
              return (
                <OverflowMenuItem
                  className="b-workflow-card__option"
                  requireTitle={false}
                  onClick={option.onClick}
                  itemText={option.itemText}
                  primaryFocus={option.primaryFocus}
                  key={option.itemText}
                />
              );
            })}
          </OverflowMenu>
        </div>
        <div className="c-workflow-card__info">
          <div className="c-workflow-card__icon">
            <img className="b-workflow-card__icon" src={imgs[workflow.icon]} alt="icon" />
          </div>
          <div className="c-workflow-card__description">
            <h2 className="b-workflow-card__name">{workflow.name}</h2>
            <p className="b-workflow-card__description">{workflow.description}</p>
            <span data-tip data-for={workflow.id} className="b-workflow-card-launch">
              <AlertModal
                ModalTrigger={() => (
                  <img src={playButton} className="b-workflow-card-launch__icon" alt="Execute workflow" />
                )}
                modalContent={closeModal => (
                  <ConfirmModal
                    title="Execute workflow?"
                    subTitleTop="It will run"
                    closeModal={closeModal}
                    affirmativeAction={this.handleExecute}
                    affirmativeText="Run"
                    theme="bmrg-white"
                  />
                )}
              />
            </span>
            <Tooltip id={workflow.id} place="bottom" theme="bmrg-white">
              Execute workflow
            </Tooltip>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(WorkflowCard);
