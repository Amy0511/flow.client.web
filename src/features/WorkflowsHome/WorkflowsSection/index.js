import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import Button from "@boomerang/boomerang-components/lib/Button";
import Tooltip from "@boomerang/boomerang-components/lib/Tooltip";
import WorkflowCard from "./WorkflowCard";
import "./styles.scss";

class WorkflowSection extends Component {
  static propTypes = {
    team: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    searchQuery: PropTypes.string.isRequired,
    updateWorkflows: PropTypes.func.isRequired
  };

  render() {
    const { team, history, searchQuery } = this.props;
    let workflows = [];
    if (searchQuery) {
      workflows = team.workflows.filter(workflow => workflow.name.toLowerCase().includes(searchQuery.toLowerCase()));
    } else {
      workflows = team.workflows;
    }

    if (workflows.length > 0) {
      return (
        <div className="c-workflow-section">
          <div className="c-workflow-section__header">
            <label className="b-workflow-section__team">{team.name}</label>
          </div>
          <div className="c-workflow-section__workflows">
            {workflows.map(workflow => (
              <WorkflowCard
                workflow={workflow}
                updateWorkflows={this.props.updateWorkflows}
                teamId={team.id}
                key={workflow.id}
              />
            ))}
            <Button className="b-workflow-placeholder" onClick={() => history.push(`/creator/overview`)}>
              <div className="b-workflow-placeholder__box">
                <div data-tip data-for={team.id} className="b-workflow-placeholder__text">
                  +
                </div>
              </div>
              <Tooltip theme="bmrg-white" id={team.id}>
                Create Workflow
              </Tooltip>
            </Button>
          </div>
        </div>
      );
    }

    return null;
  }
}

export default withRouter(WorkflowSection);
