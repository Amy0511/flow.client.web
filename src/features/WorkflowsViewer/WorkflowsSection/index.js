import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import Button from "@boomerang/boomerang-components/lib/Button";
import WorkflowCard from "./WorkflowCard";
import "./styles.scss";

class WorkflowSection extends Component {
  static propTypes = {
    team: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    searchQuery: PropTypes.string.isRequired,
    updateWorkflows: PropTypes.func.isRequired
  };

  render() {
    const { team, history, searchQuery } = this.props;
    let workflows = [];
    if(searchQuery) {
      workflows = team.workflows.filter(workflow => workflow.name.toLowerCase().includes(searchQuery.toLowerCase()));
    } else {
      workflows = team.workflows;
    }

    if(workflows.length > 0) {
      return(
        <div className="c-workflow-section">
          <div className="c-workflow-section__header">
            <label className="b-workflow-section__team">{team.name}</label>
            <Button onClick={()=> history.push(`/editor`)} theme="bmrg-black">
              Create Workflow
            </Button>
          </div>
          <div className="c-workflow-section__workflows">
            {
              workflows.map(workflow=>{
                return <WorkflowCard workflow={workflow} updateWorkflows={this.props.updateWorkflows} teamId={team.id}/>
              })
            }
          </div>
        </div>
      );
    }

    return null;
  }
}

export default withRouter(WorkflowSection);
