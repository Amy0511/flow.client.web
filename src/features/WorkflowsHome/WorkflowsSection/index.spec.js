import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import WorkflowsSection from "./index";

const mockfn = jest.fn();

const searchQuery = "test";
const setActiveTeamAndRedirect = mockfn;
const updateWorkflows = mockfn;
const setActiveTeam = mockfn;
const executeWorkflow = mockfn;
const deleteWorkflow = mockfn;
const team = {
  id: "1234",
  name: "Lucas' team",
  workflows: [
    {
      id: "hQDkX9v",
      name: "lucas-workflow-1",
      description: "blablabla",
      status: "published",
      icon: "utility"
    },
    {
      id: "456",
      name: "lucas-workflow-2",
      description: "blablabla",
      status: "published",
      icon: "utility"
    },
    {
      id: "789",
      name: "lucas-workflow-3",
      description: "blablabla",
      status: "draft",
      icon: "secure"
    }
  ]
};
const history = {};

describe("WorkflowsSection --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowsSection", () => {
    const renderedValue = renderer.create(
      <MemoryRouter>
        <WorkflowsSection
          history={history}
          searchQuery={searchQuery}
          team={team}
          updateWorkflows={updateWorkflows}
          executeWorkflow={executeWorkflow}
          deleteWorkflow={deleteWorkflow}
          setActiveTeam={setActiveTeam}
          setActiveTeamAndRedirect={setActiveTeamAndRedirect}
        />
      </MemoryRouter>
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowsSection --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <WorkflowsSection
          history={history}
          searchQuery={searchQuery}
          team={team}
          updateWorkflows={updateWorkflows}
          executeWorkflow={executeWorkflow}
          deleteWorkflow={deleteWorkflow}
          setActiveTeam={setActiveTeam}
          setActiveTeamAndRedirect={setActiveTeamAndRedirect}
        />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
