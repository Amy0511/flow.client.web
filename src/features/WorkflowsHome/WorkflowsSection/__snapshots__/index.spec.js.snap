// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowsSection --- Snapshot Capturing Snapshot of WorkflowsSection 1`] = `
<div
  className="c-workflow-section"
>
  <div
    className="c-workflow-section__header"
  >
    <label
      className="b-workflow-section__team"
    >
      Lucas' team
    </label>
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      style={
        Object {
          "width": "fit-content",
        }
      }
    >
      <svg
        alt="Provide a description that will be used as the title"
        aria-label="Provide a description that will be used as the title"
        className="b-workflow-import__icon"
        data-for="Lucas' team"
        data-tip={true}
        fillRule="evenodd"
        height="24"
        name="icon--upload"
        role="img"
        viewBox="0 0 24 24"
        width="24"
      >
        <title>
          Provide a description that will be used as the title
        </title>
        <path
          d="M5 6.6l1.2 1.1L11 3v13h2V3l4.8 4.7L19 6.6 12 0z"
        />
        <path
          d="M22 14v6H2v-6H0v10h24V14z"
        />
      </svg>
    </div>
    <div
      className="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
      id="Lucas' team"
    >
      Import Workflow
    </div>
  </div>
  <div
    className="c-workflow-section__workflows"
  >
    <button
      className="b-workflow-placeholder"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="b-workflow-placeholder__box"
      >
        <div
          className="b-workflow-placeholder__text"
          data-for="1234"
          data-tip={true}
        >
          +
        </div>
      </div>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
        id="1234"
      >
        Create Workflow
      </div>
    </button>
  </div>
</div>
`;
